src/ut_kvp.c

```diff
@@ -34,19 +35,32 @@
 ut_kvp_instance_t *gKVP_Instance = NULL;
 
 #define UT_KVP_MAGIC (0xdeadbeef)
+#define UT_KVP_MAX_INCLUDE_DEPTH 5
 
 typedef struct
 {
     uint32_t magic;
     struct fy_document *fy_handle;
 } ut_kvp_instance_internal_t;
 
+// Struct to store the downloaded data
+typedef struct
+{
+   char *memory;
+    size_t size;
+} ut_kvp_download_memory_internal_t;
```

Comment:
doesn't need ut_kvp prefix if it's internal of course :-)

Reply:
yeah, true that. However, tried to follow the same standard across the file.